### PR TITLE
MWP 1.1 update on iOS - introduce ActionSource, expose SDK.Error enum type, version check

### PR DESCRIPTION
**Pull request by bangtoven**

### CODE REVIEW

File: ios/CoinbaseWalletSDK/CoinbaseWalletSDK.swift [hunk 1]
```diff
@@ -88,7 +105,7 @@ public final class CoinbaseWalletSDK {
             return unsupportedHandShakeMethod.contains(where: {action.method == $0 })
         })
         
-        guard hasUnsupportedAction != true else {
+        guard !(CoinbaseWalletSDK.verificationMethod == .callbackConfirmation && hasUnsupportedAction == true) else {
```
Comment (bangtoven) [inline ios/CoinbaseWalletSDK/CoinbaseWalletSDK.swift]: we don't need `hasUnsupportedAction` restrictions for public key verification mode

File: ios/CoinbaseWalletSDK/Message/Request/RequestMessage.swift [hunk 2]
```diff
@@ -9,7 +9,7 @@ import Foundation
 
 @available(iOS 13.0, *)
 public enum RequestContent: UnencryptedContent {
-    case handshake(appId: String, callback: URL, initialActions: [Action]?)
+    case handshake(appId: String, callback: URL, initialActions: [Action]?, verificationMethod: VerificationMethod?)
```
Comment (bangtoven) [inline ios/CoinbaseWalletSDK/Message/Request/RequestMessage.swift]: making it as optional to ensure backward compatibility with older wallet/dapps

File: ios/CoinbaseWalletSDK/Error.swift [hunk 3]
```diff
@@ -7,9 +7,8 @@
 
 import Foundation
 
-@available(iOS 13.0, *)
 extension CoinbaseWalletSDK {
-    enum Error: Swift.Error {
+    public enum Error: Swift.Error {
```
Comment (bangtoven) [inline ios/CoinbaseWalletSDK/Error.swift:11]: image

**Review by vishnumad — APPROVED** [review]
LGTM!

**Pull request merged**